File: src/lib/models/DTO/Domain/BaseDomainDto.cs
using System;

namespace DTO.Domain
{
    public class BaseDomainDto
    {
        public long Id{ get; set; }
        public Guid UniqueId{ get; set; }
        public DateTime CreationDateUtc{ get; set; }
        public long CreationUserId{ get; set; } = 0;
        public bool Inactive{ get; set; }

        public BaseDomainDto() =>
            (this.Id, this.UniqueId, this.CreationUserId, this.CreationDateUtc, this.Inactive) = 
            (0, Guid.NewGuid(), 0, DateTime.UtcNow, false);

        public bool Validate(bool throwException = true)
        {
            bool invalid = this.UniqueId == Guid.Empty || this.CreationDateUtc == null;
            if(invalid && throwException) throw new Exception($"Invalid {nameof(BaseDomainDto)} structure");
            return invalid;
        }
    }
}

File: src/ui/asq-ui/src/app/_models/domain/base-domain-dto.ts
import * as uuid from 'uuid';
import * as moment from 'moment';

export class BaseDomainDto{
    public id: number = 0;
    public uniqueId: string = uuid.v4();
    public creationDateUtc: moment.Moment = moment.utc();
    public creationUserId: number = 0;
    public inactive: boolean = false;

    constructor(){ }
}
